kotlin/internal/jvm/compile.bzl

```diff
@@ -389,37 +514,34 @@ def _run_kt_builder_action(ctx, rule_kind, toolchains, dirs, srcs, friend, compi
     )
 
     ctx.actions.run(
-        mnemonic = "KotlinCompile",
+        mnemonic = mnemonic,
         inputs = depset(
-            ctx.files.srcs,
+            ctx.files.srcs + generated_src_jars,
             transitive = [compile_deps.compile_jars, transitive_runtime_jars],
         ),
         tools = tools,
         input_manifests = input_manifests,
         outputs = [f for f in outputs.values()],
         executable = toolchains.kt.kotlinbuilder.files_to_run.executable,
-        execution_requirements = {"supports-workers": "1"},
+        execution_requirements = {
+            "supports-workers": "1",
+            "worker-key-mnemonic": "KotlinCompile"
+        },
         arguments = [args],
         progress_message = progress_message,
         env = {
             "LC_CTYPE": "en_US.UTF-8",  # For Java source files
         },
     )
 
-def kt_jvm_produce_jar_actions(ctx, rule_kind):
+def kt_jvm_produce_output_jar_actions(ctx, rule_kind):
```

Comment:
I should rename this method or maybe just inline or refactor it into the final output jar merge section above.

Reply:
Inlined / removed this method.